### ISSUE TITLE
Can't login on android device - twitter_login: ^4.0.1

### ISSUE THREAD

**Issue by scorpiotiftif** [0maru/twitter_login#69]
Good evening,
I contact you, because I can't connect to twitter following all the procedures and settings that can be found on the internet and stackoverflow.

Creation of the Twitter developer account : **OK**
Generation of the API key : **OK**
Generation of the API secret key : **OK**
Activation of 3-legged OAuth : **is enabled**
Added callback URL(s) : **DONE**

- mangateck://
- twitter://
- twittersdk://
- https://mangateck-18b6f.firebaseapp.com/__/auth/handler

Request email address from users is **disabled/enabled** => nothing changes

Permission: I tried everything => **Read, Write, and Direct Messages** 

<img width="850" alt="Capture d’écran 2021-12-10 à 23 02 55" src="https://user-images.githubusercontent.com/13355593/145647279-def7c1fb-5fec-481e-8570-6a4ae9bc4e98.png">

-------------------------

```
Future<UserCredential> signInWithTwitter() async {
    // Create a TwitterLogin instance
    final twitterLogin = TwitterLogin(
        apiKey: 'XXXXXXX',
        apiSecretKey:'YYYYYYYYYYYYYYY',
        redirectURI: 'twitter://'
    );

    // Trigger the sign-in flow
    final authResult = await twitterLogin.login();
    switch (authResult.status!) {
      case TwitterLoginStatus.loggedIn:
        print("SUCCESS");
        break;
      case TwitterLoginStatus.cancelledByUser:
        print("CANCELLED BY USER");
        break;
      case TwitterLoginStatus.error:
        print("ERROR");
        print(authResult.status!.index); // RESULT ==> 2
        print(authResult.status!); // RESULT ==> TwitterLoginStatus.error
        print(authResult.errorMessage); // RESULT ==> Exception: Exception: HttpException: Failed Forbidden
        break;
    }

    // Create a credential from the access token
    final twitterAuthCredential = TwitterAuthProvider.credential(
      accessToken: authResult.authToken!, // This code break but I'm not here
      secret: authResult.authTokenSecret!,
    );

    // Once signed in, return the UserCredential
    return await FirebaseAuth.instance.signInWithCredential(twitterAuthCredential);
  }
```

------ MANIFEST -------
```
<intent-filter>
                <action android:name="android.intent.action.VIEW" />

                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <!-- Accepts URIs that begin with "example://gizmos” -->
                <!-- Registered Callback URLs in TwitterApp -->
                <data android:scheme="twitter" /> <!-- host is option -->
            </intent-filter>
```

  -------------------------
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6 20G165 darwin-x64, locale fr-FR)
    • Flutter version 2.5.3 at /Users/francoisgaucher/FlutterSDK/202102_flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (8 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/francoisgaucher/Library/Android/sdk
    • Platform android-31, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio Preview.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.10.2

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 4.2 Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio 2.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 4.2 Preview 2.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[✓] Android Studio
    • Android Studio at /Users/francoisgaucher/Downloads/Android Studio 4.2 Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] Connected device (2 available)
    • SM N986B (mobile) • R5CN80J7RDJ • android-arm64 • Android 11 (API 30)
    • macOS (desktop)   • macos       • darwin-x64    • macOS 11.6 20G165 darwin-x64

• No issues found!

**Comment by scorpiotiftif** [comment]
I add an information, the authorization seems to be valid by twitter on the "authorization" side because I find the application on the "User" side in the profile->Security and connection->Application and session-> application connected->"APP".

<img width="595" alt="Capture d’écran 2021-12-11 à 09 13 56" src="https://user-images.githubusercontent.com/13355593/145669763-9c92108a-a70b-4a99-8b02-e766a80c2fc1.png">

I regenerated several times the different keys to test also.

For information: the authentication by google & facebook (nothing to see) works on the same basis of the code (the problem must come from a configuration without doubt).

Translated with www.DeepL.com/Translator (free version)

**Comment by scorpiotiftif** [comment]
I hadn't tested it on ios yet, but I did. (Iphone X => iOS 15.0)
The result is the same:

flutter: Connect to TWITTER
flutter: ERROR
flutter: 2
flutter: TwitterLoginStatus.error
flutter: Exception: Exception: HttpException: Failed Forbidden


<img width="409" alt="Capture d’écran 2021-12-11 à 11 30 06" src="https://user-images.githubusercontent.com/13355593/145673375-1fa3f3a6-cd10-4d51-96df-887b6d0d6dc5.png">

**Comment by 0maru** [comment]
@scorpiotiftif 
The plugin does not seem to work if you create a Twitter developer account after 15 November 2021.
When did you create your developer account?

Also, I am currently working on TwitterAPIv2 support.

https://github.com/0maru/twitter_login/issues/67

**Comment by scorpiotiftif** [comment]
This would explain that... I just created it 3 days ago.

Thanks for the analysis. I look forward to the next version of the plugin then.

Do you have a time range on the integration of this new api ?

**Comment by 0maru** [comment]
@scorpiotiftif 
I will be releasing v4.1.0-dev today in Japan time.

**Comment by 0maru** [comment]
@scorpiotiftif 
I have released v4.1.0-dev.
Twitter API v2.0 is now supported.
Please update this plugin.

**Comment by scorpiotiftif** [comment]
Hello @0maru 
I'm updating my project and test the preview version.

**Comment by rikai-trongpq** [comment]
> @scorpiotiftif I have released v4.1.0-dev. Twitter API v2.0 is now supported. Please update this plugin and change it to use the loginV2 method.

Can you update the doc to use your package? I stuck `Redirecting you back to the application. This may take a few moments.` .

**Comment by rikai-trongpq** [comment]
![Untitled](https://user-images.githubusercontent.com/83201151/146158182-eae9cf3d-ed2e-49be-9319-8b959de6e326.png)
This is my config in Twitter developers.
and this is my AndroidManifest
```
<intent-filter>
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data android:scheme="@string/fb_login_protocol_scheme" /> [This is config for another function]
                <data android:scheme="twitterkit-xxxxxxxxxxxxxxxxxxxxx" />

            </intent-filter>
```

**Comment by scorpiotiftif** [comment]
Hi !
Thank's a lot about this new version that work like a charm !
I had encounter an other problem but not due to your plugin.

PS: if someone read this maybe it will be usefull if you want to complete register/login with **FIREBASE** : on twitter developer account, upgrade OR choose the "ELEVATED" product NOT "ESSENTIAL" otherwise you can't login through "twitter_login" AND "firebase" !

**Comment by scorpiotiftif** [comment]
Just an other things.... think to update README to specify to use "loginV2" method in some case

**Comment by rikai-trongpq** [comment]
> Just an other things.... think to update README to specify to use "loginV2" method in some case

Hey bro. Can u show me your code, manifest and config on Twitter developers to implement it? Of course you should cover important information.

**Comment by matheszabi** [comment]
It is closed because:
"I found the issue: I just needed to relaunch the Twitter app which was launched, and then the redirect worked fine :)"
But mine is not solved: Android fails to come back to app. It hangs on: "redirecting you back to the application. This may take a few moments" .  Waiting more than 10 minutes and it is still there.

I have re-created the twitter app several times.
Used v1.1 only, then changed to 2.0 only. Not solved.

The iOS Simulator comes back to  the app.
The iPhone device (iOS 15.5) comes back to app.
